addon/services/panel-actions.js

```diff
@@ -21,29 +21,35 @@ const State = EmberObject.extend({
   group: null
 });
 
-export default Service.extend({
-  _registry: EmberObject.create({
-    keys: A([]),
+const Registry = EmberObject.extend({
+  keys: computed(function() {
+    return A([]);
+  }),
 
-    unknownProperty: function(name) {
-      const state = State.create();
+  // probably not too safe, should only be used in tests
+  reset() {
+    this.get('keys')
+      .map(i => i) // copy, so we dont mess with binding/loops
+      .forEach((key) => {
+        delete this[key];
+      });
 
-      this.get('keys').addObject(name);
-      this.set(name, state);
+    this.get('keys').clear();
+  },
+});
 
-      return state;
-    },
+export default Service.extend({
+  _registry: computed(function() {
```

Comment:
I think setting this up in `init` would be a bit easier to reason about, the cost is very low here so making it eager shouldn't affect too much.

I'm thinking something like:

```js
export default Service.extend({
  init() {
    this._super(...arguments);

    this._registry = Registry.create();
  }
});
```